### ISSUE TITLE
CONAN_EXPORTED variable in CMake

### ISSUE THREAD

**Issue by ovidiub13** [conan-io/conan#2408]
Acording to the docs: http://docs.conan.io/en/latest/changelog.html?highlight=conan_exported
the CMake variable `CONAN_EXPORT` should be set when running `conan create`, and not set when running `conan build`.
The documentation is actually non existant for this variable and the only mention is in the changelog, saying on version `0.19.0`: 
```
Feature: new cmake CONAN_EXPORTED variable, can be used in CMakeLists.txt to differentiate building in the local conan cache as package and building in user space
```
From my tests I see that the variable is set in both scenarios. I've tested this with Conan 1.0.4.
Please let me know if I'm using this wrong.

I'm testing it like this:
```
if(CONAN_EXPORTED) # set when running "conan build"
    message("=================================== CONAN_EXPORT is set: ${CONAN_EXPORT}")
else()
    message("=================================== CONAN_EXPORT is not set: ${CONAN_EXPORT}")
endif()
```
In both scenarios the output is: 
```
=================================== CONAN_EXPORT is set:
```

----
To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by memsharded** [comment]
You are outputting ``${CONAN_EXPORT}`` while the check is for ``CONAN_EXPORTED``

**Comment by ovidiub13** [comment]
I've corrected my code:
```
if(CONAN_EXPORTED) # set when running "conan build"
    message("=================================== CONAN_EXPORTED is set: ${CONAN_EXPORTED}")
else()
    message("=================================== CONAN_EXPORTED is not set: ${CONAN_EXPORTED}")
endif()
```

My output is now:
```
=================================== CONAN_EXPORTED is set: 1
```
in both situations.

**Comment by memsharded** [comment]
Oh, yes, I am sorry, now I see the issue.

The actual meaning of ``CONAN_EXPORTED`` is that cmake is being launched by the ``CMake`` helper from the ``build()``, method, i.e. managed by conan. Versus doing manually the ``cmake ...`` commands over the same CMakeLists.txt. It does not refer to the location (user folder/cache), and unfortunately we cannot rename that misleading name, cause it would be breaking.

**Comment by ovidiub13** [comment]
Since it's not documented, and only mentioned in the changelog, I believe the breaking wouldn't be that big.
Also, this means that this is not a good solution to differentiate between the two processes. Is there another?

**Comment by lasote** [comment]
Maybe we could create a new variable in CMake to follow the `conanfile.in_local_cache` is that the differentiation you want to make?

http://docs.conan.io/en/latest/reference/conanfile/attributes.html?highlight=in_local_cache#in-local-cache

**Comment by ovidiub13** [comment]
Yes, that is the differentiation I wanted to make. I can now set a variable myself, now that I know of `conanfile.in_local_cache`. I'll test this one now.

**Comment by memsharded** [comment]
I am fine too creating a CONAN_IN_LOCAL_CACHE cmake variable automatically.

**Comment by danimtb** [comment]
closed by conan-io/docs#817
